src/extension.ts

```diff
@@ -36,15 +41,126 @@ export function activate(context: ExtensionContext) {
+    } else {
+        return paths
+    }
+}
+
+// originally from https://github.com/viperproject/viper-ide/blob/master/client/src/Settings.ts
+function normalise(tools: string, path: string | PlatformDependentPath) : string {
+    if (typeof path !== 'string') {
+        // handle object values
+        if (isMac && path.mac) return normalise(tools, first(path.mac))
+        else if (isLinux && path.linux) return normalise(tools, first(path.linux))
+        else throw new Error(`normalise() on an unsupported platform: ${process.platform}, or path missing`);
+    } else {
+        if (!path || path.length <= 2) return path;
+        path = path.replace(/\$viperTools\$/g, tools);
+        while (path.includes('$')) {
+            const index_of_dollar = path.indexOf('$');
+            let index_of_closing_slash = path.indexOf('/', index_of_dollar + 1);
+            if (index_of_closing_slash < 0) {
+                index_of_closing_slash = path.length;
+            }
+            const envName = path.substring(
+                index_of_dollar + 1,
+                index_of_closing_slash,
+            );
+            const envValue: string = process.env[envName] || '';
+            if (!envValue) {
+                throw new Error(
+                    `environment variable ${envName} used in path ${path} is not set`,
+                );
+            }
+            if (envValue.includes('$')) {
+                throw new Error(
+                    `environment variable ${envName} must not contain '$': ${envValue}`,
+                );
+            }
+            path = `${path.substring(
+                0,
+                index_of_dollar,
+            )}${envValue}${path.substring(index_of_closing_slash, path.length)}`;
+        }
+        return path
+    }
+}
+
 export function startServer(context: ExtensionContext) {
     // Cross-platform language server
     const module = context.asAbsolutePath(
         path.join('out', 'server', 'server.js'),
     );
+
+    let viperTools = '';
+    let java = '';
+    let serverJar = '';
+    let z3 = '';
+    const config = workspace.getConfiguration('viperSettings');
+    if (config) {
+        viperTools = normalise(viperTools, config.paths.viperToolsPath);
+        const buildVersion = config.buildVersion || 'Stable';
+        // macOS (Viper extension bugfix)
+        if (viperTools.endsWith('/Library/Application Support/Viper')) {
+            // Rewrite default directory
+            viperTools = path.resolve(
+                homedir(),
+                `Library/Application Support/Code/User/globalStorage/viper-admin.viper/${buildVersion}/ViperTools`
+            );
+        }
+        // Linux (Viper extension bugfix)
+        else if (viperTools.endsWith('/.config/Viper')) {
+            // Rewrite default directory
+            viperTools = path.resolve(
+                homedir(),
+                `.vscode-server/data/User/globalStorage/viper-admin.viper/${buildVersion}/ViperTools`
```

Comment:
```suggestion
                `.config/Code/User/globalStorage/viper-admin.viper/${buildVersion}/ViperTools`
```
Are you sure? On @crusso's setup `$HOME/.vscode-server` doesn't exist :-(

Reply:
Maybe `process.execPath` gives a hint where the config lives...

Reply:
Okay, good to know. It seems like VS Code uses a different global storage directory when using WSL, so I switched the hard-coded directory with `context.globalStorageUri.fsPath` (which is almost exactly what we're looking for).